state/settings.go

```diff
@@ -48,6 +48,30 @@ func (m settingsMap) GetBSON() (interface{}, error) {
 	return escapedMap, nil
 }
 
+// SetingsGroup models a list of Settings that can all be atomically written to
+// the backing store.
+type SettingsGroup []*Settings
+
+// Write the changes from all Settings in the group to the backing store in a
+// single transaction.
+func (sg SettingsGroup) Write() error {
+	if len(sg) == 0 {
+		return nil
+	}
+
+	var grpUpdateOps []txn.Op
+	for _, s := range sg {
+		_, ops := s.settingsUpdateOps()
```

Comment:
I would go defensive and against nil here just in case. Returning an error instead of letting it panic.

Reply:
Fixed